app/interactors/new_document/select_interactor.rb

```diff
@@ -0,0 +1,71 @@
+class NewDocument::SelectInteractor < ApplicationInteractor
+  include Rails.application.routes.url_helpers
+
+  delegate :params,
+           :user,
+           :redirect_url,
+           :document_type_selection,
+           to: :context
+
+  def call
+    find_document_type_selection
+    check_for_issues
+    find_redirect_url
```

Comment:
I've been having a bit of a think about what my thoughts are on what should go in the call method.

I think it's a reasonable shout to not use context.fail as I agree with your stance that this isn't a failing scenario.
I did think it was a pretty reasonable shout about redirect_url though I am wondering if it might actually be fine to not have to do that (and could mean losing the helper include which feels like a bit of a smell)

I also think we don't say quite enough in the call method, so my suggestion would be that we do something like:

```
find_selection
check_for_issues
return unless selected_option.document_type?
create_document
create_timeline_entry
```

I know that reintroduces the conditional that was a point of contention but I think it's ok as it is a conditional path after all. 

I guess that'd mean the controller ends up with:

```
    result = NewDocument::SelectInteractor.call(params: params, user: current_user)
    issues, document, document_type_selection, selected_option =
      result.to_h.values_at(:issues, :document, :document_type_selection, :selected_option)

    if issues
      flash.now["requirements"] = { "items" => issues.items }
      render :show,
             assigns: { issues: issues, document_type_selection: document_type_selection },
             status: :unprocessable_entity
    else
      destination = if document
                      content_path(document)
                    elsif selected_option.managed_elsewhere?
                      selected_option.managed_elsewhere_url
                    else
                      new_document_path(type: selected_option.id)
                    end
      redirect_to destination
    end
```
Which seems good to me - apologies if this is eerily similar to how you had this before.

Minor suggestions as part of that would be that `find_selection` comprises setting `context.document_type_selection` and `context.selected_option` (it feels a little unusual to me that selected option is an instance variable rather than on context); You would need to set a context.document to have that managed; `select_option.document_type?` is suggested instead of `create_document?` to avoid `create_document? create_document` and consistency `with managed_elsewhere?`.

Reply:
That's pretty much what the controller looked like before. I'll remove the last commit and revert this change.